core/domain/classifier_services_test.py

```diff
@@ -92,6 +91,76 @@ def test_string_classifier_classification(self):
             self.assertTrue(
                 self._is_string_classifier_called('dunno, just guessed'))
 
+    def test_handle_creation_of_jobs_and_mappings(self):
+        """Test the handle_jobs_creation method and handle_mappings_creation
+        method by triggering update_exploration() method.
+        """
+        exploration = exp_services.get_exploration_by_id(self.exp_id)
+        state = exploration.states['Home']
+
+        # Now, there should be no jobs in the data store.
+        all_jobs = classifier_models.ClassifierTrainingJobModel.get_all()
+        self.assertEqual(all_jobs.count(), 0)
+
+        # Modify such that job creation is triggered.
+        state.interaction.answer_groups.insert(
+            3, state.interaction.answer_groups[1])
+        answer_groups = []
+        for answer_group in state.interaction.answer_groups:
+            answer_groups.append(answer_group.to_dict())
+        change_list = [{
+            'cmd': 'edit_state_property',
+            'state_name': 'Home',
+            'property_name': 'answer_groups',
+            'new_value': answer_groups
+        }]
+        with self.swap(feconf, 'ENABLE_STRING_CLASSIFIER', True):
+            exp_services.update_exploration(
```

Comment:
@seanlip . I have a small issue here. This function currently utilises the ExplorationCommitLogs internally through get_trainable_states_dict() method. This is very flaky because the existence of ExplorationCommitLog for this version keeps varying every time I run this test. I noticed that the storage of the commit log to data store was done using a 'put_async' method. So, ExplorationCommitLogs exists sometimes and doesn't exist the other times. How can we work around this?